crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -978,12 +978,8 @@ impl SlidingSync {
     /// Handle the HTTP response.
     ///
     /// But which response? `v4::Response`, aka the Sliding Sync response, or
-    /// `SyncResponse` which still relies on `v3`?
-    /// Well that's tricky. We have both here, because this Sliding Sync
-    /// implementation is still experimental, and we didn't want to be too
-    /// invasive. Thus, `SyncResponse` doesn't support Sliding Sync yet. Hence
-    /// the fact this method handles both at the same time. It's not super
-    /// annoying but it was important to clarify that.
+    /// `SyncResponse`? We have both because `SyncResponse` doesn't support
+    /// Sliding Sync yet.
```

Comment:
What does "we have both" mean? This function takes a `v4::Response`, does it convert it?

Reply:
Oh, when it processes `v4::Response` just below, it gets a `SyncResponse`. It doesn't receive it as an argument, that's true, but the method works with both responses.

Reply:
Okay. Well I think it should be clear though, `SyncResponse` is not `v3::Response` 😅 

So I'm still not convinced about this comment..

Reply:
I'll rewrite it Monday :-).

Reply:
Done :-).